src/generator.cs

```diff
@@ -2620,8 +2620,12 @@ public void Go ()
 		foreach (Type t in types){
 			if (SkipGenerationOfType (t))
 				continue;
-
-			Generate (t);
+			try {
+				Generate (t);
+			}
+			catch (Exception e) {
+				throw new InvalidOperationException ($"When processing type {t} exception {e}");
```

Comment:
It's usually best to set the inner exception (often helps with stack traces):

```suggestion
				throw new InvalidOperationException ($"When processing type {t} exception {e}", e);
```